File: Data Structures & Algorithms/design-circular-queue/submission-0.cpp
class MyCircularQueue {
private:
    int capacity;
    vector<int> queue;
public:
    MyCircularQueue(int k) {
        capacity=k;
    }
    
    bool enQueue(int value) {
        if(queue.size()==capacity){
            return false;
        }
        queue.push_back(value);
        return true;
    }
    
    bool deQueue() {
        if(queue.empty()){
            return false;
        }
        queue.erase(queue.begin());
        return true;
    }
    
    int Front() {
        if(isEmpty()){
            return -1;
        }
        return queue[0];
    }
    
    int Rear() {
        if(isEmpty()){
            return -1;
        }
        return queue[queue.size()-1];
    }
    
    bool isEmpty() {
        if(queue.size()==0){
            return true;
        }
        return false;
    }
    
    bool isFull() {
        if(queue.size()==capacity){
            return true;
        }
        return false;
    }
};

/**
 * Your MyCircularQueue object will be instantiated and called as such:
 * MyCircularQueue* obj = new MyCircularQueue(k);
 * bool param_1 = obj->enQueue(value);
 * bool param_2 = obj->deQueue();
 * int param_3 = obj->Front();
 * int param_4 = obj->Rear();
 * bool param_5 = obj->isEmpty();
 * bool param_6 = obj->isFull();
 */

File: Data Structures & Algorithms/maximum-frequency-stack/submission-0.py
class FreqStack:

    def __init__(self):
        self.m = {}
        self.stack = {}
        self.maxs = 0

    def push(self, val: int) -> None:
        count = 1
        if val in self.m:
            count = self.m[val] +1
        self.m[val] = count
        if count > self.maxs:
            self.maxs = count
            self.stack[count] = []
        self.stack[count].append(val)
        
    def pop(self) -> int:
        res = self.stack[self.maxs].pop() 
        self.m[res] -= 1
        if not self.stack[self.maxs]:
            self.maxs -=1
        return res
        


# Your FreqStack object will be instantiated and called as such:
# obj = FreqStack()
# obj.push(val)
# param_2 = obj.pop()